### DISCUSSION TITLE
oracledb.exceptions.ProgrammingError: DPY-2007: expected a type

### DISCUSSION THREAD

**Discussion by VickyGopi** [oracle/python-oracledb#326] [General]
I am trying to execute the stored procedure using oracledb module. To start with I had set up a connection by creating dsn object for the database and created a cursor.

dsn = oracledb.makedsn(host, 1521, dbname)
connection = oracledb.connect(user="xxx", password="yyy", dsn=dsn)
cursor = connection.cursor()
cursor.callfunc('%function_name%',('%val%', '%val%', '%val%'))

when I do so I am receiving the following error whereas I am manually able to run the procedure. 

**oracledb.exceptions.ProgrammingError: DPY-2007: expected a type**

**Comment by cjbj** [comment]
Can you share more code, including the SQL to create the function?

**Comment by anthony-tuininga** [comment]
The [documentation](https://python-oracledb.readthedocs.io/en/latest/api_manual/cursor.html#Cursor.callfunc) shows that the second parameter should be a type (the return type of the function) but your example doesn't show that -- which would explain the error! If that hint doesn't help, please provide the additional information requested by @cjbj.
